plugin/builtin/helmchartgenerator/HelmChartGenerator.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// Helm chart generator
+//
+// Fetches the given chart from {ChartRepo}/{ChartName},
+// and inflates it to stdout, using the given values file.
+// This generator expects helm V3 or later.
+
+//go:generate pluginator
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/yaml"
+)
+
+// HelmChartGeneratorPlugin is a plugin to generate resources
+// from a remote helm chart.
+type HelmChartGeneratorPlugin struct {
+	h                *resmap.PluginHelpers
+	types.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	runHelmCommand   func([]string) ([]byte, error)
+	types.HelmChartArgs
+}
+
+//noinspection GoUnusedGlobalVariable
+var KustomizePlugin HelmChartGeneratorPlugin
+
+// Config uses the input plugin configurations `config` to setup the generator
+// options
+func (p *HelmChartGeneratorPlugin) Config(h *resmap.PluginHelpers, config []byte) error {
+	p.h = h
+	err := yaml.Unmarshal(config, p)
+	if err != nil {
+		return err
+	}
+	if p.ChartName == "" {
+		return fmt.Errorf("chartName cannot be empty")
+	}
+	if p.ChartRepoURL == "" {
+		return fmt.Errorf("chartRepoUrl cannot be empty")
+	}
+	p.runHelmCommand = runHelmCommand
+	return nil
+}
+
+// Generate implements generator
+func (p *HelmChartGeneratorPlugin) Generate() (resmap.ResMap, error) {
+	// check helm version. we only support V3
```

Comment:
we can merge this and add v2 later, but i think we need to support both (lacking data on migration rates).